guidance/_grammar.py

```diff
@@ -935,12 +892,12 @@ def __init__(self) -> None:
             "rx_nodes": [],
         }
         self.grammars = [self.curr_grammar]
-        self.node_id_cache: dict[GrammarFunction, int] = {}
-        self.todo: list[GrammarFunction] = []
+        self.node_id_cache: dict[ComposableGrammar, int] = {}
```

Comment:
I'm wondering if a lot of these in `LLSerializer` can actually be `GrammarFunction`s